### PR TITLE
Switch to multi arch test/images with manifests

### LINKED ISSUE #66626: Publish multi-arch images for e2e tests (Dockerfile not in test/images)
- [ ] "k8s-sample-admission-webhook" (to "gcr.io/kubernetes-e2e-test-images") 
       - fixing build for this image [here](https://github.com/kubernetes/kubernetes/pull/66665) (MERGED)
       - need to publish from directory `test/images/webhook`
- [ ] "k8s-aggregator-sample-apiserver" (to "gcr.io/kubernetes-e2e-test-images") 
       - Really old image build from k/k 1.7 branch, about to be [promoted to conformance](https://github.com/kubernetes/kubernetes/pull/63947). 
       - adding a new image [here](https://github.com/kubernetes/kubernetes/pull/66686) (APPROVED - In submit queue)
       - need to publish from directory `test/images/sample-apiserver`
- [ ] "k8s-dns-dnsmasq" (to "k8s.gcr.io") 
       - All we need is `dnsmasq`, so adding that to `dnsutils` image [here](https://github.com/kubernetes/kubernetes/pull/66695) (MERGED)
       - need to publish from directory `test/images/dnsutils`
- [ ] "gb-frontend" (to "gcr.io/google-samples" repo) 
       - looks like fixed up in https://github.com/kubernetes/examples/commit/5f484579b5cc04900591c61cf87ece00fe8426d7
       - we need https://github.com/kubernetes/examples/pull/255 merge too! (MERGED)
       - need to publish from `k8s.io/examples/guestbook/php-redis`
- [ ] "gb-redisslave" (to "gcr.io/google-samples" repo) 
       - looks like fixed up in https://github.com/kubernetes/examples/commit/5f484579b5cc04900591c61cf87ece00fe8426d7  (MERGED)
       - need to publish from `k8s.io/examples/guestbook/redis-slave`
- [ ] "nginx-slim" 
       - being fixed in https://github.com/kubernetes/kubernetes/pull/66055/ (REVIEW)

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

Recently we updated the test container images to use multi-arch fat manifests and pushed the new images to the `gcr.io/kubernetes-e2e-test-images` repository. In this changeset, we are switching to using the new images and cleaning up some of the unused image definitions from manifest.go. We are removing the folders corresponding to the unused images as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66626

**Special notes for your reviewer**:
/cc @mkumatag 
/cc @luxas 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
@spiffxp, @ixdy, @luxas - i've iterated a while on this, so as soon as this goes green, we can merge this.

**Comment by spiffxp** [comment]
/lgtm
/hold
for someone else's /approve

File: test/utils/image/manifest.go [hunk 1]
```diff
@@ -48,51 +48,45 @@ func (i *ImageConfig) SetVersion(version string) {
...
-	LogsGenerator            = ImageConfig{e2eRegistry, "logs-generator", "1.0", true}
-	Mounttest                = ImageConfig{e2eRegistry, "mounttest", "1.0", true}
-	MounttestUser            = ImageConfig{e2eRegistry, "mounttest-user", "1.0", true}
-	Nautilus                 = ImageConfig{e2eRegistry, "nautilus", "1.0", true}
-	Net                      = ImageConfig{e2eRegistry, "net", "1.0", true}
-	Netexec                  = ImageConfig{e2eRegistry, "netexec", "1.0", true}
-	Nettest                  = ImageConfig{e2eRegistry, "nettest", "1.0", true}
+	EntrypointTester         = ImageConfig{e2eRegistry, "entrypoint-tester", "1.0", false}
+	Fakegitserver            = ImageConfig{e2eRegistry, "fakegitserver", "1.0", false}
+	GBFrontend               = ImageConfig{sampleRegistry, "gb-frontend", "v5", false}
```
Comment (ixdy) [inline test/utils/image/manifest.go]: it doesn't look like the gb- images are using manifest lists yet, so setting `false` here is probably wrong.

Comment (dims) [reply test/utils/image/manifest.go]: Hmm, i eyeballed it here - https://console.cloud.google.com/gcr/images/google-samples/GLOBAL (searched there for `gb-`) let me check if something is wrong by building those images locally

Comment (dims) [reply test/utils/image/manifest.go]: you are right. i missed that. assumed that https://github.com/kubernetes/examples/commit/5f484579b5cc04900591c61cf87ece00fe8426d7 and https://github.com/kubernetes/examples/commit/5f484579b5cc04900591c61cf87ece00fe8426d7 was doing it, looks like it's not.

**Pull request merged**